lazyflow/utility/io_util/OMEZarrStore.py

```diff
@@ -0,0 +1,180 @@
+###############################################################################
+#   ilastik: interactive learning and segmentation toolkit
+#
+#       Copyright (C) 2011-2024, the ilastik developers
+#                                <team@ilastik.org>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# In addition, as a special exception, the copyright holders of
+# ilastik give you permission to combine ilastik with applets,
+# workflows and plugins which are not covered under the GNU
+# General Public License.
+#
+# See the LICENSE file for details. License information is also available
+# on the ilastik web site at:
+# 		   http://ilastik.org/license.html
+###############################################################################
+import json
+import logging
+import os
+from collections import OrderedDict
+from typing import Dict
+from urllib.parse import unquote_to_bytes
+
+import jsonschema
+import vigra
+from zarr.core import Array as ZarrArray
+from zarr.storage import FSStore, LRUStoreCache
+
+from lazyflow import rtype
+from lazyflow.utility import Timer
+from lazyflow.utility.io_util.multiscaleStore import MultiscaleStore, DEFAULT_SCALE_KEY
+
+logger = logging.getLogger(__name__)
+
+OME_ZARR_V_0_4_KWARGS = dict(dimension_separator="/", normalize_keys=False)
+OME_ZARR_V_0_1_KWARGS = dict(dimension_separator=".")
+
+
+def _get_axistags_from_spec(ome_spec: Dict) -> vigra.AxisTags:
+    if "axes" not in ome_spec:
+        # v0.1 and v0.2 did not allow variable axes
+        axis_keys = ["t", "c", "z", "y", "x"]
+    else:
+        ome_axes = ome_spec["axes"]
+        assert isinstance(ome_axes, list), f"Expected axis information to be a list, received: {ome_axes}."
+        if "name" not in ome_axes[0]:
```

Comment:
We shouldn't trust the stuff we get out of external files to be well-formed. Even if the spec says it should either be a `List[str]` or a `List[Dict]`, there's no telling what we might actually find there.

So the absolute safest thing to do would be something like:

```python
raw_axes = ome_spec.get("axes")
if not isinstance(raw_axes, list):
    # raise Exception, return Error, whatever
if all(isinstance(a, str) and a in "tczyx" for a in raw_axes):
   parsed_axes: List[str] = raw_axes
else:
    parsed_axes: List[str] = []
    for axis_obj in raw_axes:
        if not isinstance(axis_obj, dict):
            # raise eception, return err, whatever
        axis_name = axis_obj.get("name")
        if not isinstance(axis_name, str) or axis_name not in "xyztc":
            # raise, whatever
        parsed_axes.append(axis_name) 
```

Reply:
Validating the input isn't this function's job; that's up to `jsonschema.validate`. I think it's worth pointing this out in a comment, but given the scope here, I wouldn't go fully pydantic (or custom types to reflect validated JSON contents)...

Reply:
I've left it at `validate` + comments for now, though I suspect I'll be pydantifying this when I implement export to OME-Zarr, which will interact more with the metadata.